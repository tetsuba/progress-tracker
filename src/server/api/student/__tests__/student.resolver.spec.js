const { createTestServer } = require('../../../test/serverTestHelper')
const {
  StudentMockRequest,
} = require('../../../../test/mockApi/student/studentMockData')
const {
  ADD_STUDENT_MUTATION,
} = require('../../../../client/api/student/student.mutation')
const {
  STUDENTS_QUERY,
  STUDENT_QUERY,
} = require('../../../../client/api/student/student.query')

describe('student.resolver', () => {
  describe('@Query', () => {
    describe('students', () => {
      const ctx = {
        user: { id: 'userId01' },
        models: {
          Student: new StudentMockRequest(),
        },
      }

      test('should return a list of students', async () => {
        const { query } = createTestServer(ctx)
        const res = await query({
          query: STUDENTS_QUERY,
          variables: { inputs: { studentId: 'student01' } },
        })
        expect(res).toMatchSnapshot()
      })
    })
    describe('getStudent', () => {
      const ctx = {
        models: {
          Student: new StudentMockRequest(),
        },
      }

      test('should return a student if an id is found', async () => {
        const { query } = createTestServer(ctx)
        const res = await query({
          query: STUDENT_QUERY,
          variables: {
            input: {
              id: 'studentId01',
            },
          },
        })
        expect(res.data.getStudent).toMatchSnapshot()
      })

      test('should return an error if a student is not found', async () => {
        const { query } = createTestServer(ctx)
        const res = await query({
          query: STUDENT_QUERY,
          variables: {
            input: {
              id: 'studentId0',
            },
          },
        })
        expect(res.errors).toMatchSnapshot()
      })
    })
  })
  describe('@Mutation', () => {
    describe('addStudent', () => {
      const ctx = {
        user: { id: 'userId01' },
        models: {
          Student: new StudentMockRequest(),
        },
      }
      test('should add a student and return students data', async () => {
        const { mutate } = createTestServer(ctx)
        const res = await mutate({
          mutation: ADD_STUDENT_MUTATION,
          variables: {
            input: {
              firstName: 'Bill',
              lastName: 'Bee',
              DOB: '123456789',
            },
          },
        })
        expect(res).toMatchSnapshot()
      })

      test('should return an error if a student is found', async () => {
        const { mutate } = createTestServer(ctx)
        const res = await mutate({
          mutation: ADD_STUDENT_MUTATION,
          variables: {
            input: {
              firstName: 'John',
              lastName: 'Doh',
              DOB: '123456789',
            },
          },
        })
        expect(res.errors).toMatchSnapshot()
      })
    })
  })
})
