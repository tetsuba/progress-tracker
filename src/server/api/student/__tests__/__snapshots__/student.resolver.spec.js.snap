// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`student.resolver @Mutation addStudent should add a student and return students data 1`] = `
Object {
  "data": Object {
    "addStudent": Array [
      Object {
        "_id": "001",
        "firstName": "Bob",
        "lastName": "Boo",
      },
      Object {
        "_id": "studentId01",
        "firstName": "David",
        "lastName": "Blue",
      },
      Object {
        "_id": "Hello World",
        "firstName": "Bill",
        "lastName": "Bee",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`student.resolver @Mutation addStudent should return an error if a student is found 1`] = `
Array [
  [GraphQLError: STUDENT_EXISTS],
]
`;

exports[`student.resolver @Query getStudent should return a student if an id is found 1`] = `
Object {
  "assessments": Array [
    "Assessment",
    "Assessment-1",
  ],
  "firstName": "David",
  "lastName": "Blue",
}
`;

exports[`student.resolver @Query getStudent should return an error if a student is not found 1`] = `
Array [
  [GraphQLError: STUDENT_NOT_FOUND],
]
`;

exports[`student.resolver @Query students should return a list of students 1`] = `
Object {
  "data": Object {
    "students": Array [
      Object {
        "_id": "001",
        "firstName": "Bob",
        "lastName": "Boo",
      },
      Object {
        "_id": "studentId01",
        "firstName": "David",
        "lastName": "Blue",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
